crates/sparrow-compiler/src/types/inference.rs

```diff
@@ -489,6 +489,20 @@ fn least_upper_bound_data_type(a: DataType, b: &DataType) -> Option<DataType> {
         (Null, other) => Some(other.clone()),
         (other, Null) => Some(other),
 
+        // Duration types can cast to more granular duration types
+        // s -> any
+        // ms -> us,ns
+        // us -> ns
+        // nanosecond is omitted, as the catch all should find that lhs == rhs
+        (Duration(TimeUnit::Second), other @ Duration(_)) => Some(other.clone()),
```

Comment:
Could probably clean this up to be:

```
// Duration types cast to the more granular TimeUnit
(Duration(unit1), Duration(unit2)) => Some(Duration(std::cmp::max(unit1, unit2)))
```

The benefit is that that you don't need to explain as much since it's (a) fewer cases and (b) less code.

Reply:
Is the ordering of the enum an invariant we can rely on?